optimum/exporters/ipex/modeling_utils.py

```diff
@@ -27,9 +27,20 @@
 
 _IPEX_MINIMUM_VERSION_FOR_PATCHING = "2.3.0"
 
+if is_ipex_version("<", _IPEX_MINIMUM_VERSION_FOR_PATCHING):
+    raise ImportError(f"Only ipex version > {_IPEX_MINIMUM_VERSION_FOR_PATCHING} supports patching")
```

Comment:
Reply to https://github.com/huggingface/optimum-intel/pull/786#discussion_r1661330118, I moved the logic at the top of this file.

Reply:
I'm not sure we want to restrict support to IPEX >= v2.3.0 for all models (even the ones that are not patched) ?